File: ToyInterpreter/ToyInterpreter/Model/OutBuffer.cs
using System;
using System.Collections.Generic;
using ToyInterpreter.Model.MObserver;

namespace ToyInterpreter.Model
{
	/*public class OutBuffer : IOutBuffer {
		private readonly string[] arr;
		private int size;
		public OutBuffer() {
			arr = new string[Constants.stackSize];
			setSize(0);
		}

		public void add(string s) {
			arr[getSize()] = s;
			setSize(getSize() + 1);
		}
		public int getSize() {
			return size;
		}
		public void printBuffer() {
			for(int i=0; i<getSize(); ++i) Console.WriteLine(arr[i]);
		}
		private void setSize(int size) {
			this.size = size;
		}
	}*/
	[Serializable()]
	public class OutBuffer : IOutBuffer {
		private readonly List<string> arr;
		private readonly List<CObserver> o;

		public OutBuffer() {
			arr = new List<string>();
			o = new List<CObserver>();
		}

		public void addObserver(OutBufferObserver ob) {
			o.Add(ob);
		}

		private void notifyObservers() {
			for(int i = 0; i < o.Count; ++i) o[i].onUpdate(this);
		}

		public void add(string s) {
			arr.Add (s);
			notifyObservers ();
		}
		
		public int getSize() {
			return arr.Count;
		}
		public void printBuffer() {
			for(int i=0; i<getSize(); ++i) Console.WriteLine(arr[i]);
		}
		public String toString() {
			if(getSize()==0) return "OutBuffer is empty!\n";
			String r="OutBuffer:\n";
			foreach(String s in arr) 
				r+=s+"\n";
			return r;
		}
	}
}

File: ToyInterpreter 2/src/Model/OutBuffer.java
package Model;

import java.util.ArrayList;

import mobserver.OutBufferObserver;
import mobserver.StackObserver;

/*public class OutBuffer implements IOutBuffer {
	private final String[] arr;
	private int size;
	public OutBuffer() {
		arr = new String[Constants.stackSize];
		setSize(0);
	}
	
	public void add(String s) {
		arr[getSize()] = s;
		setSize(getSize() + 1);
	}
	public int getSize() {
		return size;
	}
	public void printBuffer() {
		for(int i=0; i<getSize(); ++i) System.out.println(arr[i]);
	}
	private void setSize(int size) {
		this.size = size;
	}
}*/
public class OutBuffer implements IOutBuffer,java.io.Serializable {
	private final ArrayList<String> arr;
	
	private final ArrayList<OutBufferObserver> o = new ArrayList<OutBufferObserver>();
	
	public void addObserver(OutBufferObserver ob) {
		o.add(ob);
	}
	
	private void notifyObservers() {
		for(int i = 0; i < o.size(); ++i) 
			o.get(i).onUpdate(this);
	}
	
	public OutBuffer() {
		arr = new ArrayList<String>();
	}
	
	public void add(String s) {
		arr.add(s);
		notifyObservers();
	}
	public int getSize() {
		return arr.size();
	}
	public void printBuffer() {
		for(int i=0; i<arr.size(); ++i) System.out.println(arr.get(i));
	}
	public String toString() {
		if(getSize()==0) return "OutBuffer is empty!\n";
		String r="OutBuffer:\n";
		for(String s : arr) 
			r+=s+"\n";
		return r;
	}
}